api/crud.py

```diff
@@ -193,23 +197,27 @@ async def invoice_add_related(item: models.Invoice):
     item.products = [product_id for product_id, in result if product_id]
     item.payments = []
     payment_methods = await models.PaymentMethod.query.where(models.PaymentMethod.invoice_id == item.id).gino.all()
+    currency_payment_methods = defaultdict(list)
     for method in payment_methods:
-        # TODO: multiple wallet same currency case
-        # TODO remove duplication
-        item.payments.append(
-            {
-                "payment_address": method.payment_address,
-                "payment_url": method.payment_url,
-                "rhash": method.rhash,
-                "amount": currency_table.format_currency(method.currency, method.amount),
-                "rate": currency_table.format_currency(item.currency, method.rate, fancy=False),
-                "rate_str": currency_table.format_currency(item.currency, method.rate),
-                "discount": method.discount,
-                "currency": method.currency,
-                "lightning": method.lightning,
-                "node_id": method.node_id,
-            }
-        )
+        currency_payment_methods[method.currency].append(method)
+    for currency, methods in currency_payment_methods.items():
+        for i, method in enumerate(methods, start=1):
+            name = method.currency if len(methods) == 1 else f"{currency} ({i})"
```

Comment:
I think we can apply all the formatting the frontend does right there, as name is no longer used as dictionary key.
if it is lightning and there are multiple currencies, it should look like so:
BTC (⚡) (1)
BTC (⚡) (2)

Or do you have any better ideas?